### ISSUE TITLE
traverse_ could be on UnorderedFoldable

### ISSUE THREAD

**Issue by johnynek** [typelevel/cats#2526]
Also, we should probably implement traverse_ in terms of foldMap not foldRight
```scala
def traverse_[A, B](fa: F[A])(fn: A => G[B])(implicit G: Applicative[G]): G[Unit] =
   foldMap(fn.andThen(_.unit))(Applicative.monoid[G, Unit])
```

foldRight is not the most efficient thing in the world since it leverages Eval (although, it may be that implementation is stack safe and the above is not.

**Comment by johnynek** [comment]
Well, I guess it would have to take a `CommutativeApplicative` so maybe it is not the same function on UnorderedFoldable.

**Comment by LukaJCB** [comment]
Yeah, I think you're right in that regard, probably should also think about how #2516 plays into this.
